### PR TITLE
NMS-14185: Expose Secure Credentials Vault via Integration API

### PR DESCRIPTION
**Pull request by patrick-schweizer**
We want to use the SCV in the cloud plugin and thus need to make it available in the integration api.

Jira: https://issues.opennms.org/browse/NMS-14185

### CODE REVIEW

File: test-suites/tss-tests/src/main/java/org/opennms/integration/api/v1/timeseries/InMemoryStorage.java [hunk 1]
```diff
@@ -36,21 +36,22 @@
...
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.stream.Collectors;
 
 import com.google.re2j.Pattern;
 
 /**
- * Simulates a TimeSeriesStorage in memory (HashMap). The implementation is super simple and not very efficient.
+ * Simulates a TimeSeriesStorage in memory (ConcurrentHashMap). The implementation is super simple and not very efficient.
```
Comment (patrick-schweizer) [inline test-suites/tss-tests/src/main/java/org/opennms/integration/api/v1/timeseries/InMemoryStorage.java:46]: unrelated to pull request. Just for clarification.

File: api/src/main/java/org/opennms/integration/api/v1/scv/SecureCredentialsVault.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.Set;
+
+import org.opennms.integration.api.v1.annotations.Consumable;
+
+@Consumable
+public interface SecureCredentialsVault {
+
+    Set<String> getAliases();
+    Credentials getCredentials(String alias);
```
Comment (patrick-schweizer) [inline api/src/main/java/org/opennms/integration/api/v1/scv/SecureCredentialsVault.java:38]: we don't restrict the access here. Technically the plugin can read all credentials. I am not sure if this ok. It comes down to if the admin trusts the plugins.

**Review by cgorantla — DISMISSED** [review]
LGTM!

**Comment by cgorantla** [comment]
May need to upgrade `SNAPSHOT` version

File: api/src/main/java/org/opennms/integration/api/v1/scv/SecureCredentialsVault.java (hunk 2 above)
Comment (patrick-schweizer) [reply api/src/main/java/org/opennms/integration/api/v1/scv/SecureCredentialsVault.java:38]: we fixed that in the implementation provided by OpenNMS

**Review by cgorantla — APPROVED** [review]

**Pull request merged**